remotes/docker/pusher.go

```diff
@@ -132,6 +145,48 @@ func (p dockerPusher) Push(ctx context.Context, desc ocispec.Descriptor) (conten
 		}
 		req.Header.Add("Content-Type", desc.MediaType)
 	} else {
+		mountCandidates := findMatchingOrigins(p.refspec, p.originProvider(desc))
+		for _, mountCandidate := range mountCandidates {
+			ctx, err := contextWithRepositoryScope(ctx, mountCandidate, false, true)
+			if err != nil {
+				// try next candidate
+				continue
```

Comment:
This would imply that `findMatchingOrigins` returned something malformatted, I think? It could be argued that this should be fatal and the error returned, but I think at least it should log an error.